bc/core/tests.py

```diff
@@ -94,7 +96,71 @@ def test_truncate_descriptions(self):
             result, _ = template.format(**test)
             self.assertTrue(
                 len(result) <= template.max_characters,
-                msg="Failed with dict: %s.\n"
-                "%s is longer than %s"
+                msg="Failed with dict: %s.\n%s is longer than %s"
                 % (test, result, template.max_characters),
             )
+
+
+class TextImageTest(TestCase):
+    tests = (
+        # Simple case
+        {"title": "short title", "description": 10 * "short description"},
+        # Short title and long description
+        {"title": 5 * "short title", "description": 20 * "short description"},
+        # Long title and long description
+        {"title": 15 * "short title", "description": 20 * "short description"},
+        {"title": 30 * "short title", "description": 50 * "short description"},
+    )
+
+    def test_creates_canvas_smaller_than_max_width(self):
+        for test in self.tests:
+            instance = TextImage(**test)
+            width, _ = instance.get_initial_dimensions()
+            self.assertLessEqual(
+                width,
+                instance.max_width,
+                msg="Failed with dict: %s.\n%s is larger than %s"
+                % (test, width, instance.max_width),
+            )
+
+    def test_max_character_count_avoids_overflow(self):
+        EDGE_CASE = (
+            {
+                "title": (
+                    "Case: Braidwood Management v. Becerra (ACA prev. care"
+                    " challenge)"
+                ),
+                "description": (
+                    "Transcript Order Form: re 115 Notice of Appeal,,,,"
+                    " transcript not requested Reminder: If the transcript is"
+                    " ordered for an appeal, Appellant must also file a copy"
+                    " of the order form with the appeals court. (Lynch,"
+                    " Christopher) (Entered: 04/10/2023)"
+                ),
+            },
+        )
+
+        for test in self.tests + EDGE_CASE:
```

Comment:
Do you need to run the `self.tests` here too? If so, it feels like you should just combine these into one test b/c they're so similar. If not, then two tests is fine, but `self.test` should go into the `test_creates_canvas_smaller_than_max_width`.